src/koheesio/pandas/categorical_encoding.py

```diff
@@ -0,0 +1,57 @@
+from sklearn.preprocessing import OneHotEncoder
+import pandas as pd
+from pydantic import BaseModel
+from typing import List, Optional
+from koheesio.steps import Step
+
+from typing import List, Dict
+import pandas as pd
+from pydantic import BaseModel
+
+class PandasCategoricalEncoding(BaseModel):
+    """
+    Encodes categorical columns using one-hot encoding or ordinal encoding.
+
+    Attributes
+    ----------
+    columns : List[str]
+        List of categorical columns to encode.
+    encoding_type : str, optional
+        Type of encoding, either "one-hot" or "ordinal" (default is "one-hot").
+    drop_first : bool, optional
+        Whether to drop the first dummy column for one-hot encoding (default is True).
+    ordinal_mapping : dict, optional
+        A dictionary mapping categorical values to integers for ordinal encoding.
+    """
+
+    columns: List[str]
+    encoding_type: str = "one-hot"
+    drop_first: bool = True
+    ordinal_mapping: Dict[str, Dict] = None
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        if self.encoding_type not in ['one-hot', 'ordinal']:
+            raise ValueError(f"Invalid encoding type: {self.encoding_type}. Expected 'one-hot' or 'ordinal'.")
+
+    def execute(self, data: pd.DataFrame) -> pd.DataFrame:
+        """
+        Executes the categorical encoding transformation on the provided dataset.
+
+        Parameters
+        ----------
+        data : pd.DataFrame
+            The input dataset to encode.
+
+        Returns
+        -------
+        pd.DataFrame
+            The dataset with the specified categorical columns encoded.
+        """
+        if self.encoding_type == 'one-hot':
+            data = pd.get_dummies(data, columns=self.columns, drop_first=self.drop_first)
+        elif self.encoding_type == 'ordinal':
+            for column in self.columns:
+                if column in data.columns and self.ordinal_mapping and column in self.ordinal_mapping:
+                    data[column] = data[column].map(self.ordinal_mapping[column]).fillna(-1).astype(int)
+        return data
```

Comment:
a few things about `execute` (for when you change to `Step`:
- execute takes no arguments, instead at the DataFrame as one of the input Fields; add something like this `df: Optional[pd.DataFrame] = Field(default=None, description="...")` - I will explain why you want this as Optional in a bit
- execute is expected to deal with input (your Fields) and generate Output
- this Output does not need to be returned explicitly, the Step parent-class takes care of this.
- instead, add a `.transform` method that can take a DataFrame as input

This means you can change your code like this:
1. add an `Output` class
2. add a `.transform` method
3. update your `execute` method accordingly

Should look something like this (of course add docstrings and things like that):
```python
class PandasCategoricalEncoding(PandasStep):
    ...
    class Output(PandasStep.Output):
        df: pd.DataFrame = Field(..., description="output pandas DataFrame"
    
    def transform(self, df: Optional[pd.DataFrame]):
        self.df = df or self.df
        if not self.df:
            raise RuntimeError("No valid Dataframe was passed")
        self.execute()
        return self.output.df

    def execute(self) -> Output:
        if self.encoding_type == 'one-hot':
            self.output.df = pd.get_dummies(self.df, columns=self.columns, drop_first=self.drop_first)
        elif self.encoding_type == 'ordinal':
            data = self.df
            for column in self.columns:
               for column in self.columns:
                if column in data.columns and self.ordinal_mapping and column in self.ordinal_mapping:
                    data[column] = data[column].map(self.ordinal_mapping[column]).fillna(-1).astype(int)
            self.output.df = data
```

This way you can interact with your class this way:
```python
encoding_step = PandasCategoricalEncoding(
    columns=["color"],
    encoding_type="one-hot",
    drop_first=False  # Adjusted to match expected columns
)
encoded_data = encoding_step.transform(self.data)
```

This way this interface matches what we do for Spark. 
Note: I will work on making Transformation base classes for Pandas in a separate PR.

For reference: https://github.com/Nike-Inc/koheesio/blob/9bd29ec553cc6306520926a89d5e6405641b5994/src/koheesio/spark/transformations/__init__.py#L35-L192

Reply:
making `df` an `Optional` type allows us to either give the `df` as an argument when initializing the class, or pass it through `transform` - this is exactly how we do it inside the Spark module at the moment